### PR TITLE
Added error handling

### PR DESCRIPTION
**Pull request by kirkbushell**
Moves the JSON api error handling from the flarum/core project to this one. Includes tests.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 composer.lock
 phpunit.xml
 vendor
+.idea/
```
Comment (GrahamCampbell) [inline .gitignore]: :-1:

File: .idea/php.xml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (GrahamCampbell) [inline .idea/php.xml]: oops

File: .gitignore (hunk 1 above)
Comment (kirkbushell) [inline .gitignore]: Adding files that you wish to be ignored?

Comment (GrahamCampbell) [inline .gitignore]: that belongs in the global ignore file

Comment (kirkbushell) [inline .gitignore]: That's the kind of feedback I was hoping for, instead of just a thumbs down. Bit more constructive - thanks.

**Comment by tobscure** [comment]
Can you please send this to the 0.2 branch?

**Comment by kirkbushell** [comment]
Yeah sure. There's still a bit that needs to come from this into this project, once the json-api has been updated.

**Pull request closed without merging**